osvr/RenderKit/RenderManagerBase.cpp

```diff
@@ -1433,188 +1463,407 @@ namespace renderkit {
-        /// back from room space.
-        q_xyz_quat_type q_worldFromRoom;
-        makeIdentity(q_worldFromRoom);
-        if (inWorldSpace && (params.worldFromRoomAppend != nullptr)) {
-            q_from_OSVR(q_worldFromRoom, *params.worldFromRoomAppend);
-        }
-        q_xyz_quat_type q_worldFromEye;
-        q_xyz_quat_compose(&q_worldFromEye, &q_worldFromRoom, &q_roomFromEye);
-
-        /// Invert the above matrices, to produce eyeFromWorld.
-        q_xyz_quat_type q_eyeFromWorld;
-        q_xyz_quat_invert(&q_eyeFromWorld, &q_worldFromEye);
-
-        /// Include the impact of the space we're rendering to.
-        /// This is spaceFromRoom; put on the right and multiply it on
-        /// the left by the above inverted matrix.  (If we are going
-        /// into one of these spaces, worldFromRoom will be the
-        /// identity so we don't need to invert and reapply it.)
-        q_xyz_quat_type q_worldFromSpace;
-        if (inWorldSpace) {
-            makeIdentity(q_worldFromSpace);
-        } else {
-            OSVR_TimeValue timestamp;
-            if (osvrGetPoseState(
-                    m_callbacks[whichSpace].m_interface, &timestamp,
-                    &m_callbacks[whichSpace].m_state) == OSVR_RETURN_FAILURE) {
-                // They asked for a space that does not exist.  Return false to
-                // let them know we didn't get the one they wanted.
-                return false;
-            }
-            q_from_OSVR(q_worldFromSpace, m_callbacks[whichSpace].m_state);
-        }
-        q_xyz_quat_type q_eyeFromSpace;
-        q_xyz_quat_compose(&q_eyeFromSpace, &q_eyeFromWorld, &q_worldFromSpace);
-
-        /// Store the result into the output pose
-        OSVR_from_q(eyeFromSpace, q_eyeFromSpace);
-        return true;
+		// If eye poses are present, use them instead of head +- IPD
+		bool hasEyes = hasLeftEyePose() && hasRightEyePose();
+		if (!hasEyes)
+		{
+			/// We need to determine the transformation that takes points
+			/// in the space we're going to render in and moves them into
+			/// the space described by the projection matrix.  That space
+			/// is eye space, which has the eye at the origin, X to the
+			/// right, Y up, and Z pointing into the camera (opposite to
+			/// the viewing direction).
+
+			/// Include the impact of rotating the screen around the
+			// eye location for HMDs who have this feature.  This is
+			// computed in terms of the percent overlap of the screen.
+			// We rotate each eye away from the other by half of the
+			// amount they should not overlap.  NOTE: This assumes
+			// that both eyes are at the same location w.r.t. the
+			// overlap percent.
+			// @todo Verify this assumption.
+			// @todo should we be doing this transformation when we have eye poses, or would that be included in the pose?
+			q_xyz_quat_type q_rotatedEyeFromEye;
+			makeIdentity(q_rotatedEyeFromEye);
+			double rotateEyesApart = 0;
+			double overlapFrac =
```

Comment:
It looks like eye rotation to match overlap has been lost in the copied code before.
I'd rather see a more surgical if() that just replaces the eye-location part and leaves the rest of the code intact so we can't get code drift in the two paths.